pkg/controller/framework/controller.go

```diff
@@ -119,13 +154,8 @@ func (c *Controller) Requeue(obj interface{}) error {
 func (c *Controller) processLoop() {
 	for {
 		obj := c.config.Queue.Pop()
-		err := c.config.Process(obj)
-		if err != nil {
-			if c.config.RetryOnError {
-				// This is the safe way to re-enqueue.
-				c.config.Queue.AddIfNotPresent(obj)
-			}
-		}
+		// TODO: Dispatch it correctly.
+		c.workQueue[0] <- obj
```

Comment:
@lavalamp - regarding our discussion... I think that we can't use workQueue here, because IIUC it can potentially merge operations into a single one [e.g. add + update = different add] or what is even worse add + delete = nothing. Here we clearly need to process all actions.